torch/csrc/jit/codegen/cuda/scheduler/compile_time_info.h

```diff
@@ -86,6 +88,16 @@ class VectorizableInputsAndOutputs {
       CompileTimeEntryType::VECTORIZABLE_INPUTS_AND_OUTPUTS;
 };
 
+//! Entry type definition class for `VECTORIZE_MAPS`,
+//!  stores the vectorizable TensorViews on a fusion's inputs and outputs.
+class VectorizeMaps {
+ public:
+  using DataType =
+      std::vector<vectorize_helper::ContiguousInnerDimensionsMapper>;
+  static const CompileTimeEntryType EntryType =
+      CompileTimeEntryType::VECTORIZE_MAPS;
+};
```

Comment:
According to https://github.com/csarofeen/pytorch/pull/2124#discussion_r1005043035, can we just cache a std::vector<Val*> for vector sizes? If so, could these cached sizes be used for vectorization validation?

Reply:
Vectorize maps are relative to a reference. We'd have to cache the reference too, and then if for some reason the vectorization was run an other the validation wouldn't work... yeah we might be able to do something but I think it'd be nicer to just wait until we have better indexing logic we can process.

Reply:
Note to self: vectorize maps being relative to a reference is actually a bit of a pain point in the end. However, I don't have a good solution. The issue I'm hitting right now is the reference has an inner most broadcast dimension, but that dimension is concretized in the fusion. Our logic actually doesn't detect that we should vectorize those dimensions but not the reference, instead it tries to vectorize a more inner dimension in the reference which obviously results in vectorize size of 1 (or it definitely should since otherwise it would be an issue).